src/Microsoft.NetCore.Analyzers/Core/Security/SslProtocolsAnalyzer.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.NetCore.Analyzers.Security.Helpers;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class SslProtocolsAnalyzer : DiagnosticAnalyzer
+    {
+        internal static DiagnosticDescriptor DeprecatedRule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA5397",
+            nameof(MicrosoftNetCoreAnalyzersResources.DeprecatedSslProtocolsTitle),
+            nameof(MicrosoftNetCoreAnalyzersResources.DeprecatedSslProtocolsMessage),
+            descriptionResourceStringName: nameof(MicrosoftNetCoreAnalyzersResources.DeprecatedSslProtocolsDescription),
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca5397",
+            customTags: WellKnownDiagnosticTags.Telemetry);
+        internal static DiagnosticDescriptor HardcodedRule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA5398",
+            nameof(MicrosoftNetCoreAnalyzersResources.HardcodedSslProtocolsTitle),
+            nameof(MicrosoftNetCoreAnalyzersResources.HardcodedSslProtocolsMessage),
+            descriptionResourceStringName: nameof(MicrosoftNetCoreAnalyzersResources.HardcodedSslProtocolsDescription),
+            isEnabledByDefault: false,
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca5398",
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        private readonly ImmutableHashSet<string> HardcodedSslProtocolsMetadataNames = ImmutableHashSet.Create(
+            StringComparer.Ordinal,
+            "Tls12",
+            "Tls13");
+
+        private const int UnsafeBits = 12 | 48 | 192 | 768;    // SslProtocols Ssl2 Ssl3 Tls10 Tls11
```

Comment:
Probably should be ignored, but I prefer to see binary addition in hex (0x4 | 0x8 | 0x30 | 0xc0 | 0x300) because the math is easier.

Reply:
If they were flags that we defined, I would agree, but in this case, I'd rather just copy the values that the documentation says, which is in decimal.